src/Corax/IndexEntry/IndexEntryWriter.cs

```diff
@@ -371,26 +371,81 @@ public void Write<TEnumerator>(int field, TEnumerator values, IndexEntryFieldTyp
         _dataIndex = dataLocation;
     }
 
+
+    [MethodImpl(MethodImplOptions.AggressiveInlining)]
     private int WriteNullsTableIfRequired<TEnumerator>(TEnumerator values, int dataLocation, ref IndexEntryFieldType indexEntryFieldLocation)
         where TEnumerator : IReadOnlySpanIndexer
+    {        
+        // If we have an small number of values, we just allocate the space for the nulls table in the stack. 
+        if (values.Length < 32 * 8)
+        {            
+            return WriteNullsTableIfRequiredStackalloc(values, dataLocation, ref indexEntryFieldLocation);
+        }
+
+        // If it is big enough, we will just allocate it on the heap and most likely will end up outside of the hot-path when
+        // the JIT compiler aggresively optimize it. 
+        return WriteNullsTableIfRequiredHeap(values, dataLocation, ref indexEntryFieldLocation);
+    }
+
+    private int WriteNullsTableIfRequiredStackalloc<TEnumerator>(TEnumerator values, int dataLocation, ref IndexEntryFieldType indexEntryFieldLocation)
+        where TEnumerator : IReadOnlySpanIndexer
     {
-        // We will include null values if there are nulls to be stored.           
         int nullBitStreamSize = values.Length / (sizeof(byte) * 8) + (values.Length % (sizeof(byte) * 8) == 0 ? 0 : 1);
-        byte* nullStream = stackalloc byte[nullBitStreamSize];
+        Debug.Assert(nullBitStreamSize < 64, "The maximum reasonable to allocate here is 64 bytes.");
+
+        byte* nullBitStreamBuffer = stackalloc byte[nullBitStreamSize];        
```

Comment:
Skip all of this. Allocate the constant value always.
Cheap and doesn't worth our time to actually compute this.
We should still assert on the size, of course.

Reply:
Isn't it still unbounded?

Reply:
No, we would be wasting CPU time as this will emit the following code regardless if it is a fixed value or a calculated value:

```asm
00007FFEB8FDB69E  push        0  
00007FFEB8FDB6A0  push        0  
00007FFEB8FDB6A2  dec         rax  
00007FFEB8FDB6A5  jne         00007FFEB8FDB69E  
```

This means that this will create on the get go `8 bytes` of stack to deal with the usual and common case of up to 64 items. Bigger than 64 items will iterate to generate as much stack space as possible. The branch misprediction in all cases required to deal with the loop end would be much more costly than the `div` and `sar` operations required to compute the size which most of the times is `0` for the purpose of the `jne` instruction.